battlepass/src/types.rs

```diff
@@ -3,32 +3,35 @@ use codec::MaxEncodedLen;
 
 #[derive(Encode, Decode, Clone, PartialEq, Eq, PartialOrd, Ord, TypeInfo, MaxEncodedLen)]
 pub enum BattlepassState {
-	Draft = 0,
-	Active = 1,
-	Closed = 2,
+	DRAFT = 0,
+	ACTIVE = 1,
+	CLOSED = 2,
 }
 impl Default for BattlepassState {
 	fn default() -> Self {
-		Self::Draft
+		Self::DRAFT
 	}
 }
 
 #[derive(Encode, Decode, Default, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
-pub struct Battlepass<Hash, AccountId, BlockNumber, BoundedString> {
+/// Battlepass struct
+///
+/// `collection_id`: Collection that will store all claimed Battlepass-NFTs
+pub struct Battlepass<Hash, AccountId, BoundedString> {
 	pub creator: AccountId,
 	pub org_id: Hash,
 	pub name: BoundedString,
 	pub cid: BoundedString,
-    pub season: u32,
-    pub price: u16,				// TODO: introduce currency
+	pub season: u32,
+	pub price: u16,				// TODO: introduce currency
 	/// Collection that will store all claimed Battlepass-NFTs
-	pub collection_id: u32,
-	pub created: BlockNumber,
-	pub mutated: BlockNumber,
+	pub collection_id: u32
```

Comment:
true, actually we can just use one collection per battlepass, organization instead of more maps

Reply:
For each Battlepass season we may have multiple collections to store different types of minted NFTs. E.g. one collection to store claimed Battlepass-NFTs and separate collections for each type of Reward.

Reply:
why not store all related in one collection?

Reply:
We can of course. But if we store different types of rewards in different collections, we can configure collection properties for each type. For example, Reward#1 can be claimed only 10 times, Reward#2 - unlimited times. So we create a separate collection per each reward with different NFT limits.